lib/acl_threadsupport/test/acl_threadsupport_test.cpp

```diff
@@ -95,7 +95,7 @@ volatile int threadvar = 0;
 
 void *test_thread(void *) {
   while (1) {
-    while (threadtest_state == SERVER_STATE)
+    while (check_threadtest_state(SERVER_STATE, true))
```

Comment:
In general, factoring out common functionality in a separate function is commendable, but in this case the mutex-wrapped code is a single line and the test becomes harder to grasp upon first sight. I would suggest to simply repeat the mutex-locking code; the fundamental issue here is not the code repetition, but that `threadtest_state` can be accessed without a mutex in the first place! (Is there a C++17 construct that could be used to make the mutex mandatory? Not for this pull request, just a thought for later.)

What do you think of something like the following to replace the above loop?
```c++
while (true) {
  acl_mutex_lock(&mymutex);
  if (threadtest_state != SERVER_STATE) break;
  acl_mutex_unlock(&mymutex);
  acl_thread_yield();
}
```

Reply:
Thanks for the suggestion :) I removed the helper function and change the while loops as suggested accordingly.